src/lib/queries/contest/getContestLeaderboard.ts

```diff
@@ -6,26 +6,58 @@ import getServerQueryClient from '@/server/utils/queryClient'
 import {withApiErrorHandler} from '@/lib/utils/common/error'
 import {
   ContestLeaderboard,
+  GetContestLeaderboardParams,
   getContestLeaderboard,
+  getContestLeaderboardCount,
 } from '@/server/actions/contest/getContestLeaderboard'
 
-const getQueryOptions = (contestId: string | undefined) => ({
-  queryKey: queryKeys.contests.leaderboard(contestId).queryKey,
+const getContestLeaderboardQueryOptions = (
+  params: GetContestLeaderboardParams,
+) => ({
+  queryKey: queryKeys.contests.leaderboard(params).queryKey,
   queryFn: withApiErrorHandler(() => {
-    if (!contestId) {
+    if (!params.contestId) {
       throw new Error('Contest ID is required.')
     }
 
-    return getContestLeaderboard({contestId})
+    return getContestLeaderboard(params)
   }),
 })
 
 export const useGetContestLeaderboard = (
-  contestId: string | undefined,
+  params: GetContestLeaderboardParams,
   options?: Partial<UseQueryOptions<ContestLeaderboard>>,
-) => useQuery({...getQueryOptions(contestId), ...options})
+) => useQuery({...getContestLeaderboardQueryOptions(params), ...options})
+
+export const prefetchGetContestLeaderboard = async (
+  params: GetContestLeaderboardParams,
+) => {
+  const queryClient = getServerQueryClient()
+  await queryClient.prefetchQuery(getContestLeaderboardQueryOptions(params))
+}
+
+const getContestLeaderboardCountQueryOptions = (
+  contestId: string | undefined,
+) => ({
+  queryKey: queryKeys.contests.leaderboardCount(contestId).queryKey,
+  queryFn: withApiErrorHandler(() => {
+    if (!contestId) {
+      throw new Error('Contest ID is required.')
+    }
+
+    return getContestLeaderboardCount(contestId)
+  }),
+})
+
+export const useGetContestLeaderboardCount = (
+  contestId: string | undefined,
+  options?: Partial<UseQueryOptions<{count: number}>>,
+) =>
+  useQuery({...getContestLeaderboardCountQueryOptions(contestId), ...options})
```

Comment:
This is actually something that we'll need for every paginated response, so I'm thinking we could use some standard paginated response that would already include the total count, for example:
```
type PaginationParams = {
  totalCount: number,
  offset: number,
  limit: number,
  ...
}  

type PaginatedResponse<T> = {
  data: T,
  pageParams: PaginationParams,
}  
```

Could be added to TODO tasks and implemented later 🤷

Reply:
I'm not sure if I understand 🤔 Let's talk about it further and add it to the list of TODOs 👍